db/db_iterator_test.cc

```diff
@@ -18,9 +18,34 @@
 
 namespace rocksdb {
 
-class DBIteratorTest : public DBTestBase {
```

Comment:
I reintroduced a bug that we fixed before and the db_iterator_test seems passing. Would be great if we add a test that would catch the bug if reoccurs? or even better if they are existing tests that could be extended to catch the bug.

```
--- a/utilities/transactions/write_prepared_txn.cc
+++ b/utilities/transactions/write_prepared_txn.cc
@@ -59,7 +59,7 @@ Iterator* WritePreparedTxn::GetIterator(const ReadOptions& options,
   Iterator* db_iter = wpt_db_->NewIterator(options, column_family);
   assert(db_iter);

-  return write_batch_.NewIteratorWithBase(column_family, db_iter);
+  return write_batch_.NewIteratorWithBase(db_iter);
```

Reply:
This bug is on a higher layer. We need to add the test to write_prepared_transaction_test.cc.